lib/types/ip.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package common
+package types
```

Comment:
maybe, but i don't feel these type definitions are actually adding any additional function - as it happens I'm currently looking into whether we can do something about handling our own marshaling etc. and different ways to validate ... upshot of which is that we could probably do away with all of these additional types - so a little loathe to keep splitting up at the moment.